amqp-impl/src/test/java/io/streamnative/pulsar/handlers/amqp/test/frame/AmqpClientChannel.java

```diff
@@ -32,6 +32,17 @@ public void add(Object response) {
     }
 
     public Object poll() {
-        return responses.poll();
+        int count = 3;
+        Object object = responses.poll();
+        while (object == null && count > 0) {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            object = responses.poll();
+            count--;
+        }
+        return object;
```

Comment:
Confuse there, it just polling the response, why we need to add the specific logic?

Reply:
The polling can be null because our methods are async, the method had not completed in the channel when the poll request is coming.

Reply:
Ok, we can use `queue.poll(timeout)`.

Reply:
OK.